### PR TITLE
Current resource view

### PR DESCRIPTION
**Pull request by SergeyGaluzo**
Adds current resource view for backwards compatibility of the database schema enhancements in PR https://github.com/microsoft/fhir-server/pull/4731

This PR has to be deployed before PR https://github.com/microsoft/fhir-server/pull/4731

https://microsofthealth.visualstudio.com/Health/_workitems/edit/135369

### CODE REVIEW

**Review by mikaelweave — DISMISSED** [review]
Approved with a question.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql [hunk 1]
```diff
@@ -1,4 +1,22 @@
-﻿CREATE TABLE dbo.Resource
+﻿CREATE TABLE dbo.CurrentResource -- This is replaced by view CurrentResource
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql:1]: qq: why is this created and dropped straight away as a table?

Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql]: I put elaborate comment in the code.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Views/CurrentResource.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE VIEW dbo.CurrentResource
```
[the remark this answers is not in the document]
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Views/CurrentResource.sql:1]: No. We deal with soft deletes with separated filters in WHERE clause.

**Review by mikaelweave — APPROVED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**